### PR TITLE
Add naming-convention fitness function diagnostics

### LINKED ISSUE #778: 3.5.5: Naming Convention Fitness Functions
## Summary

Implement 3 info-level advisory rules for DDD naming conventions. All are heuristic-based (regex, not NLP) — info-level, never block CI. Depends on #774 (3.5.1).

### `EVENT_NOT_PAST_TENSE`

Detection: For each event in all clusters, flag if name ends with `ing` (case-insensitive). Heuristic — some false positives at boundary cases are acceptable.

Examples:
- `OrderCreated` → pass
- `OrderPlaced` → pass
- `OrderCreating` → flag ✗
- `OrderProcessing` → flag ✗
- `OrderStatus` → pass (noun, not -ing)

Fix: "Rename to past tense (e.g., `OrderCreating` → `OrderCreated`)"

---

### `COMMAND_NOT_IMPERATIVE`

Detection: For each command in all clusters, flag if name does not start with a recognized imperative verb prefix (case-insensitive):

```python
IMPERATIVE_VERBS = {
    "Create", "Update", "Delete", "Cancel", "Submit", "Approve", "Reject",
    "Register", "Activate", "Deactivate", "Process", "Start", "Stop",
    "Complete", "Archive", "Restore", "Send", "Publish", "Generate",
    "Confirm", "Place", "Add", "Remove", "Set", "Reset", "Mark", "Assign",
    "Transfer", "Convert", "Merge", "Split", "Validate", "Request",
}
```

Examples:
- `CreateOrder` → pass
- `PlaceOrder` → pass
- `OrderCreation` → flag ✗
- `OrderCommand` → flag ✗

Fix: "Rename to imperative verb form (e.g., `OrderCreation` → `CreateOrder`)"

---

### `AGGREGATE_NOT_NOUN`

Detection: For each non-abstract aggregate, flag if name ends with process/capability suffixes:
`-ing`, `-er`, `-or`, `-ator`, `-ible`, `-ful`, `-ive`, `-ous`

Examples:
- `Order` → pass
- `Customer` → pass (ends in -er but is clearly a domain noun — consider excluding -er from this list or accepting false negatives)
- `OrderProcessor` → flag ✗ (ends in -or)
- `OrderProcessing` → flag ✗ (ends in -ing)
- `OrderManager` → flag ✗ (ends in -er)

**Implementation note**: `-er` suffix is common for domain nouns too (`Customer`, `User`, `Supplier`). Consider only flagging if the name doesn't appear in a domain concept whitelist, OR only flag `-ing`, `-or`, `-ator`, `-ive`, `-ous` and leave `-er` out to reduce false positives.

Fix: "Rename to a domain concept noun (e.g., `OrderProcessor` → `Order`)"

---

## Suppression

Teams with legacy naming should use `disabled_rules = ["EVENT_NOT_PAST_TENSE"]` in config rather than per-element suppression.

## Tests

Parametrized test cases for each rule covering: clearly valid (pass), clearly invalid (flag), and boundary cases:
- `EVENT_NOT_PAST_TENSE`: `OrderCreated`, `OrderPlaced`, `OrderCancelled` → pass; `OrderCreating`, `OrderProcessing` → flag; `OrderStatus` → pass
- `COMMAND_NOT_IMPERATIVE`: `CreateOrder`, `PlaceOrder` → pass; `OrderCreation`, `OrderCommand` → flag
- `AGGREGATE_NOT_NOUN`: `Order`, `Customer` → pass; `OrderProcessor`, `OrderProcessing` → flag

## Key Files

- `src/protean/ir/builder.py`
- `tests/ir/test_elements_contracts_diagnostics.py`

Closes #773 (partial)

### PR DESCRIPTION
**Pull request by subhashb**
## Summary
- Add three info-level advisory naming diagnostics to the IR builder: `EVENT_NOT_PAST_TENSE` (gerund-named events), `COMMAND_NOT_IMPERATIVE` (commands not starting with a verb from a pinned `IMPERATIVE_VERBS` set), and `AGGREGATE_NOT_NOUN` (aggregates ending in a verb/gerund/adjective suffix)
- All three are `level: "info"`, so they never gate CI under the default `[lint].level = "warn"` floor, and they honor the existing per-element `suppress_checks` / `[lint].suppressions` channels
- Verb and suffix sets are aligned to the sets pinned in the issue (restores `-ive/-ous/-able/-ible/-ful` adjective suffixes and ten imperative verbs, drops `-ate/-ates` to avoid false-flagging nouns like `State`/`Certificate`)

## Test plan
- [x] `test_elements_contracts_diagnostics.py` covers all three rules, including framework-managed event skip, agent-noun exclusions, and the pinned verb/suffix regression cases
- [ ] Core tests pass (`protean test`)

Closes #778

### CODE REVIEW

File: src/protean/ir/builder.py [hunk 1]
```diff
@@ -2284,6 +2350,139 @@ def _diagnose_email_deprecated(self, ir: dict[str, Any]) -> None:
...
+            ),
+            "fix": (
+                "Rename the command to a verb-first imperative phrase "
+                "(e.g. `PlaceOrder`)."
+            ),
+        }
+        for cluster in ir["clusters"].values():
+            for command in cluster["commands"].values():
+                name = command["name"].lower()
+                if not any(name.startswith(verb) for verb in IMPERATIVE_VERBS):
```
[the remark this answers is not in the document]
Comment (subhashb) [reply src/protean/ir/builder.py]: Fixed. `COMMAND_NOT_IMPERATIVE` now requires a CamelCase token boundary right after the matched verb — the character following the verb prefix must be uppercase (or the name must end there) — so `AddressChange` no longer false-passes as starting with `Add`, while `AddItem`/`Add` still correctly match. Added a regression test (`test_verb_prefix_requires_camelcase_boundary`) asserting `AddressChange` is flagged and `AddItem` is not.
